### ISSUE TITLE
Holding CTRL+V spams clipboard imports

### ISSUE THREAD

**Issue by SilvyPaws** [Yellow-Dog-Man/Resonite-Issues#3402]
### Describe the bug?

I found when on desktop, if you hold CTRL+V it'll spam the heck out of import windows. 

### To Reproduce

Have an image in your clipboard and just hold CTRL+V

### Expected behavior

For the prompt to only appear once. You can make it appear by letting go of one of the keys and press it again if you need another import. 

### Resonite Version Number

Beta 2024.12.18.477

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Desktop

### Log Files

N/A

### Reporters

SilvyPaws

**Comment by epicEaston197** [comment]
This is not a bug this is intentional....   I hate it too

**Comment by Dusty-Sprinkles** [comment]
Seems like something to put on a toggle (feature req for setting instead of a bug?) at least, I rarely need to import something multiple times, and if I do I'd just duplicate it ingame

**Comment by AmasterAmaster** [comment]
yea, why would one need to request to spam a clipboard in the first place?? >.> Sure in a keyboard ui focus that would make sense, but to spam slot spawning in a world would be a major problem.

**Comment by Psychpsyo** [comment]
> yea, why would one need to request to spam a clipboard in the first place?? >.> Sure in a keyboard ui focus that would make sense, but to spam slot spawning in a world would be a major problem.

Nobody requested spamming of the clipboard importer.

**Comment by Frooxius** [comment]
This is definitely not a bug.

If you hold a key, it will start getting "repeated", which is normal behavior, not just in Resonite, but also in OS (Windows/Linux) general.

E.g. if you copy a chunk of text and then hold Ctrl+V, it will start getting pasted repeatedly. This happens outside of Resonite too - e.g. even in the text box here on GitHub. This is exact same thing happening.

I don't really see why would this need a technical solution and engineering hours spent on this. If you want want this to happen... just don't hold Ctrl+V?

**Comment by Dusty-Sprinkles** [comment]
Its pretty easy to accidentally double-import at the very least, I've done it more than a few times, and I don't think its really that common in non-text contexts. If you're pasting something really large it could be a problem too with lag

**Comment by Frooxius** [comment]
How come it happens so easily on your end? This isn't really something that I've done on accident a lot. Usually I just tap Ctrl+V once and that's it. I don't see a scenario where this is happening very commonly, unless you have your windows repeat key delay set really low.

**Comment by Dusty-Sprinkles** [comment]
the repeat delay does not have meaningful numbers but its like, here for me

![Image](https://github.com/user-attachments/assets/7f970796-4367-4cfd-92a5-cd24a9d7fc88)

**Comment by Frooxius** [comment]
Have you tried increasing it?

**Comment by Dusty-Sprinkles** [comment]
I could but I kinda like how it is right now for non-resonite applications

though I think my issue here is less important personally cause I'd be happy with just some kind of setting to adjust pasting behaviour personally

**Comment by art0007i** [comment]
> Have you tried increasing it?

actually changing the keyboard settings in windows does not affect inside of resonite.
- #3226

I do think that it would be good to have an option to disable repeating specifically on the in-world pasting still. I don't really see a reason you'd want to spawn multiple import dialogs for the same thing by holding down ctrl+v

**Comment by epicEaston197** [comment]
> I do think that it would be good to have an option to disable repeating specifically on the in-world pasting still.

Unfortunately Resonite Just uses Ctrl+V as a generic Pasting input to my knowledge and there's nothing to distinguish in world pasting versus pasting something into a text field 

I might be extremely wrong about this though

**Comment by art0007i** [comment]
> Unfortunately Resonite Just uses Ctrl+V as a generic Pasting input to my knowledge and there's nothing to distinguish in world pasting versus pasting something into a text field
>
> I might be extremely wrong about this though

there is some stuff to distinguish between pasting in world and in text. even if there wasn't, then it could be added to the game to solve this request.

**Comment by shiftyscales** [comment]
It would seem to me the best solution here would be #3226 to ensure user's preference in repeat delay is respected- users unintentionally repeating key presses could adjust that value to their need and it would implicitly resolve this issue.

**Comment by epicEaston197** [comment]
> It would seem to me the best solution here would be [#3226](https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/3226) to ensure user's preference in repeat delay is respected- users unintentionally repeating key presses could adjust that value to their need and it would implicitly resolve this issue.

Yeah that might fix that but I still think it's not very intuitive if you hold down the key for it to paste multiple times. What are you gonna do with twenty five images of pumpkin pie? You only need one image of pumpkin pie for your pumpkin pie cube. Now you have to delete the other importers. It just doesn't make sense to me.

**Comment by art0007i** [comment]
I agree with easton. I think pasting to import an asset is basically never needed to be done multiple times. if you need the same thing many times you can import once, then duplicate, instead of spawning many import dialogs

**Comment by AmasterAmaster** [comment]
If it were a separate setting to mass import the same thing over and over again, I think that would be fine. But yea duplication in-platform is usually the way to go to handle that.

**Comment by SilvyPaws** [comment]
Here's why it's such a problem for me, I use push to talk a ton, holding V is very normal, when snapped to objects while using push to talk, it's a plague because you need to hold ctrl to move your camera around, especially when doing protoflux lol so sometimes you're mid sentence, hold ctrl to move then suddenly there's 30 import windows you now have to individually click through haha

**Comment by Dusty-Sprinkles** [comment]
Not sure why this issue was closed but it probably should be reopened, not sure if silvy decided to stop playing and closed their issues as a result but even still it does affect players who do want to play still

**Comment by Dusty-Sprinkles** [comment]
@DanteTucker @shiftyscales This one still needs to be reopened
